lib/bootloader_setup.pm

```diff
@@ -380,7 +380,89 @@ sub uefi_bootmenu_params {
     # assume bios+grub+anim already waited in start.sh
     # in grub2 it's tricky to set the screen resolution
     send_key 'e';
-    assert_screen("grub2-enter-edit-mode", 30) if is_jeos;
+    assert_screen("grub2-enter-edit-mode", 30) if (is_jeos || is_usb_boot);
+
+    # Workaround for sle micro 6 baremetal installation with SelfInstall ISO on USB.
+    # See details in https://bugzilla.suse.com/show_bug.cgi?id=1218095#c69.
+    # - remove from kernel command line `console=tty0`, to let installation shown in sol
+    # - add in kernel command line `rd.kiwi.term=linux`, to let installation dialog
+    #   have proper background color
+    # - set serial console in kernel command line
+    if (is_ipmi && is_selfinstall && is_usb_boot) {
+        my $counter = 0;
+        my $max_tries = 5;
+        while (!check_screen('no-tty0-but-term-linux', 2) && $counter++ < $max_tries) {
+            # Re-enter grub edit by discarding changes for 2+ round
+            if ($counter > 1) {
+                send_key_until_needlematch('bootloader-grub2', 'esc', 3, 2);
+                send_key_until_needlematch('grub2-enter-edit-mode', 'e', 3, 2);
+            }
+            record_info("Doing round $counter of grub editing...");
+
+            # Go to linux line
+            # For efficiency, use send_key with wait_screen_change
+            send_key('down', wait_screen_change => 1) for (1 .. 3);
+            # To mitigate sol unstability, use below while loop.
+            # Can't use send_key_until_needlematch because it will die.
+            my $_counter = 0;
+            my $_max = 5;
+            while (!check_screen('grub2-edit-linux-line', 2) && $_counter++ < $_max) {
+                send_key('down', wait_screen_change => 1);
+            }
+            next if (!check_screen('grub2-edit-linux-line', 2));
+
+            # Move cursor to `console=ttyS0`
+            send_key('right', wait_screen_change => 1) for (1 .. 60);
+            $_counter = 0;
+            $_max = 10;
+            while (!check_screen('on-linux-console-ttyS0', 2) && $_counter++ < $_max) {
+                send_key('right', wait_screen_change => 1);
+            }
+            next if (!check_screen('on-linux-console-ttyS0', 2));
+
+            # Delete `ttyS0`
+            send_key('delete', wait_screen_change => 1) for (1 .. 5);
```

Comment:
The smallest change in default boot params will break this code.

Reply:
Yes and no. Firstly,  indeed such modification depends on the kernel line settings, but which grub editing isn't? Secondly, the needles are already made as modular as possible, little depending on other parameters. Last, based on discussions in the bug 1218095 which drove hard to make these changes in DEFAULT ISO, but no way. So the DEFAULT ISO settings won't be easy to be changed. So, IMO, this is acceptable practical solution based on the situation, although not ideal indeed.

But if I overlook any other reason supporting your statement, please help share and welcome practical solutions suggestions :)

Reply:
I was hoping to do that way which would be much simpler ;-) . I do not like current way either, which is very cumbersome. Two reasons stop me:
- Not all machines support 'end' key which can make cursor go to the end of the line, eg HPE Gen 10
- That's theory and what it expects to work. See bug here https://bugzilla.suse.com/show_bug.cgi?id=1217541#c30. When some meaningless setting is put at somewhere that it shall neglect, but it is causing problems now. So better not let this happen again at designing.

Reply:
@mdoucha We'd like to have this PR merged soon to catch up with SLE Micro 6 RC1 test. Would you please have a check if it is fine for you now?